File: Data Structure/8. Tree/binary-tree/binaryTree.py
# a binary tree node class
class Node:
  def __init__(self, data):
    self.data = data
    self.left = None
    self.right = None

# bfs algorithm
def printLevelOrder(root):
  # base case
  if root == None:
    return
  
  # create an empty list for level order traversal
  queue = []
  queue.append(root) # enqueue root and initailize height

  while len(queue):
    temp = queue[0]
    print(str(temp.data) + " "),
    queue.remove(queue[0])

    # add left child 
    if temp.left: queue.append(temp.left)
    if temp.right: queue.append(temp.right)

root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.right.right = Node(5)

print("Level order traversal(BFS) of binary tree is")
printLevelOrder(root)

File: Data Structures/8. Tree/binary-tree/binaryTree.cpp
#include <iostream>
#include <queue>
using namespace std;

// A Binary Tree Node
struct Node {
  int data;
  struct Node* left;
  struct Node* right;
  // node constructor
  Node(int value): data(value), left(NULL), right(NULL) {}
};

// create a new node
Node* newNode(int data) {
  return new Node(data);
}

// Iterative method to find height of Bianry Tree
void printLevelOrder(Node *root) {
  // Base Case
  if (root == NULL)
    return;

  // Create an empty queue for level order traversal
  queue<Node *> q;

  // Enqueue Root and initialize height
  q.push(root);

  while (q.empty() == false) {
    // Print front of queue and remove it from queue
    Node *node = q.front();
    cout << node->data << " ";
    q.pop();

    // Enqueue left child 
    if (node->left != NULL)
      q.push(node->left);

    // Enqueue right child
    if (node->right != NULL)
      q.push(node->right);
  }
}

// Driver program to test above functions
int main() {
  // Let us create binary tree shown in above diagram
  Node *root = newNode(1);
  root->left = newNode(2);
  root->right = newNode(3);
  root->left->left = newNode(4);
  root->left->right = newNode(5);

  cout << "Level Order traversal(BFS) of binary tree is \n";
  printLevelOrder(root);
  return 0;
}